### PR TITLE
static_website_tls produce deprecation warnings with cohttp>=6.0.0

### PR DESCRIPTION
**Pull request by samoht**

### CODE REVIEW

**Comment by hannesm** [comment]
Thanks for your work.

Unfortunately, the whole OpamVersionCompare is rather unintuitive (at least I always struggle with it):
```OCaml
# OpamVersionCompare.compare "6.0.0~foo" "6.0.0";;
- : int = -1
# OpamVersionCompare.compare "6.0.0~foo" "6.0";;
- : int = 1
# OpamVersionCompare.compare "6.0.0~foo" "6";;
- : int = 1
```

And the cohttp_server / cohttp_client "devices" in mirage (mirage_impl_http) already emit a `~max:"6.0.0"`. This used to work fine until people started to release pre-releases to opam-repository (or, some of the tooling (looking at opam-monorepo) didn't adhere to "avoid-version").

Now, how to deal with it? Maybe the path forward is to (a) adjust the bounds in the mirage utility (to use "6" instead of "6.0.0"), and then cut a mirage release rather sooner than later (still, there are some PRs pending that would benefit from review/comments).

File: applications/static_website_tls/config.ml [hunk 1]
```diff
@@ -22,7 +22,9 @@ let https_port =
   Key.(create "https_port" Arg.(opt ~stage:`Run int 4433 doc))
 
 let main =
-  let packages = [ package "uri"; package "magic-mime" ] in
+  let packages =
+    [ package "uri"; package "magic-mime"; package "cohttp" ~max:"6.0.0" ]
```
Comment (hannesm) [inline applications/static_website_tls/config.ml:26]:
```suggestion
    [ package "uri"; package "magic-mime"; package "cohttp" ~max:"6" ]
```

**Comment by hannesm** [comment]
and in the meantime I fixed your PR to emit the`< "6"` constraint, so CI should be happy.

**Comment by samoht** [comment]
Actually `opam-monorepo` respects `avoid-version`. The proper fix is hence: https://github.com/ocaml/opam-repository/pull/24326

**Comment by hannesm** [comment]
> Actually `opam-monorepo` respects `avoid-version`. The proper fix is hence: [ocaml/opam-repository#24326](https://github.com/ocaml/opam-repository/pull/24326)

Great to hear! Thanks for your investigation.

**Comment by hannesm** [comment]
I don't quite understand this `avoid-version`, so the cohttp-mirage 6.0.0~alpha2 was missing that flag, but this has a dependency on cohttp-lwt with =version constraint, and cohttp-lwt 6.0.0~alpha2 has a `avoid-version` flag.. now the opam-monorepo solver still decided to use cohttp-lwt in 6.0.0~alpha2?

**Comment by hannesm** [comment]
and since the main branch now succeeds, I'll close this PR. thanks again for the cross-repo investigations

**Pull request closed without merging**